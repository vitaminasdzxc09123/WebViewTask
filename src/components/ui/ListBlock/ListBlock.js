import React               from 'react';
import PropTypes           from 'prop-types';
import { Text, View }      from 'react-native';
import { useSelector }     from 'react-redux';

import Loader              from '../../base/Loader';
import { useThemedStyles } from '../../../hooks/useThemeStyles';
import AnimatedPress       from '../../base/AnimatedPress/AnimatedPress';

import Styles              from './styles';

const text = {
    buttonText : 'Check Copyright information'
}

function ListBlock({data, containerStyle, onPress, description }) {
    const { styles, colors } = useThemedStyles(Styles);
    const { item } = data || {}

    return (
        <View
           key   = { data?.index || item?.id }
           style = {[ styles.container, containerStyle ]}>
          <Text>
             {item?.name}
          </Text>
          <AnimatedPress 
               onPress = {onPress}
               style   = {styles.containerButton}>
            <Text>
               {text?.buttonText}
            </Text>
          </AnimatedPress>
         {
           Boolean(item?.copyright) && description &&
              <Text>
                 {item?.copyright}
              </Text>
         }
        </View>
    );
}

ListBlock.propTypes = {
    containerStyle : PropTypes.object
};

ListBlock.defaultProps = {
    containerStyle : {},
};

export default React.memo(ListBlock);
