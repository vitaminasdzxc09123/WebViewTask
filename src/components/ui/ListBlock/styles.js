import { StyleSheet } from 'react-native';

export default function styles({ colors, fonts }) {
    return (
        StyleSheet.create({
            container : {
                flexDirection     : 'row',
                justifyContent    : 'space-between',
                alignItems        : 'center',
                marginTop         : 10,
                flex              : 1,
                paddingHorizontal : 16,
                paddingVertical   : 16,
                borderRadius      : 12,
                backgroundColor   : colors?.BACKGROUND_GRAY
            },
            containerButton: {
                borderWidth  : 1,
                padding      : 8,
                borderRadius : 12,
                borderColor: colors?.ACCENT
            },
            title : {
                fontSize   : fonts?.font14,
                lineHeight : fonts?.font20,
                ...fonts?.regular
            },
        })
    );
}
