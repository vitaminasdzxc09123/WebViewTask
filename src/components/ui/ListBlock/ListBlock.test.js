import React from 'react';
import renderer from 'react-test-renderer';
import ListBlock from './ListBlock';

it('renders correctly', () => {
    const tree = renderer.create(<ListBlock />).toJSON();
    expect(tree).toMatchSnapshot();
  });