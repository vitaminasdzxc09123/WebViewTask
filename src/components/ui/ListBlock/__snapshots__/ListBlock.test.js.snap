// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": undefined,
        "borderRadius": 12,
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": 10,
        "paddingHorizontal": 16,
        "paddingVertical": 16,
      },
      Object {},
    ]
  }
>
  <Text />
  <View
    accessibilityState={
      Object {
        "disabled": false,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "borderColor": undefined,
          "borderRadius": 12,
          "borderWidth": 1,
          "padding": 8,
          "transform": Array [
            Object {
              "scale": 1,
            },
          ],
        }
      }
    >
      <Text>
        Check Copyright information
      </Text>
    </View>
  </View>
</View>
`;
